### DISCUSSION TITLE
Running Epoptes from a fat client

### DISCUSSION THREAD

**Discussion by gfreccia** [ltsp/ltsp#253] [Q&A]
Hi All,

I need to run Epoptes on Fat Client.
Is still valid this?
_https://epoptes.org/documentation/run-fat/_

When I run _ltsp-remoteapps dbus-launch epoptes_ I've _ltspltsp-remoteapps_ not found.

Thank you!

**Comment by alkisg** [comment]
No, it's not valid for the new LTSP.
LTSP5 remoteapps wasn't a good solution anyway, they forwarded X traffic instead of forwarding Epoptes traffic, making things slow.

I need to write documentation on how to do this with SSH socket forwarding instead...

**Comment by gfreccia** [comment]
Thanks Alkis!

So I think that for now I've to use old LTSP.
How can install it?

Giacomo Trovato

**Comment by alkisg** [comment]
Hi Giacomo,

please do not reply by email to github, see what happens:
https://github.com/ltsp/community/issues/99#issuecomment-582265965

You may click "edit" there to fix your message appearance.

For LTSP5 support, go to https://ltsp.org and then click in the "Old LTSP5 mailing lists" link. It's supported there.

**Comment by rweilert** [comment]
By the way, reading this, does it mean new LTSP doesn't support remote apps anymore, or is it just badly documented yet?
Rolf

**Comment by alkisg** [comment]
@rweilert, the new LTSP has no remoteapps implementation.

In general, remoteapps aren't specific to LTSP, so they should be implemented as a separate package that anyone with SSH/Xorg access to remote servers could use. But as I mentioned in  https://github.com/ltsp/community/issues/93#issuecomment-579869069, remoteapps won't run on Wayland and they have increasing issues from things like systemd-logind, dbus-daemon, remote notifications etc.

If someone wants to implement them as an ltsp applet, `ltsp remoteapp xxx`, no problem, patches welcome. Also, if funding is found, I'm sure someone will also be found to do the coding. :)

**Comment by rweilert** [comment]
I don't know if it would become necessary at all in my case, but it is rather probable. On LTSP5 which I am on now, e. g. Thunderbird doesn't run on the fat client, it needs to run on the server directly. Epoptes, of course. Then certain functions under Wine (everything with database, such as an address list for serial letters in Word) etc. etc... Even Midnight Commander's editor produces dead lock files if run on the fat client and not via ssh on the server directly. 

I guess the reason for all this is something like sshfs or what you listed (dbus and stuff). So if LTSP19 offers ways to overcome these restrictions, all these programs will run fine on fat clients. Otherwise it might be good to find some workaround for it.

**Comment by alkisg** [comment]
I'm not aware of any such issues, except for gnome-keyring-daemon which we reported upstream.
Thunderbird has been running fine here in LTSP5 and I just verified that it's still running fine in the new LTSP 20.
Even gnome-keyring-daemon can work fine if follow_symlinks is turned on.
And if everything else fails, there's always NFS.

In any case, the correct solution there would be to report such issues upstream, not run those apps on the server.

**Comment by gfreccia** [comment]
Hi All,

about Epoptes server, I think it's mandatory to work as remote app or you need to instruct the clients that the server is a Fat Client. But in this case the "teacher" has not the freedom to log from whatever PC.
Epoptes it's a very good tool, so I think we've to find a solution. LTSP without it it's quite lame!

Giacomo.

**Comment by alkisg** [comment]
Oh I've tested the "socket forwarding" solution and it works great, a lot better than ltsp remoteapps.
I just need to find time to write it in the epoptes documentation...

**Comment by gfreccia** [comment]
Great Alkis!
And as always... thanks for your job.

Giacomo.

**Comment by gfreccia** [comment]
Hi Alkis,

do you mean "forwarding" the connection of the server to the Fat Client?
I think I can manage it on my pfSense.
But... If the Epoptes clients open connection to server IP or "sever" DNS name,I can simply replace that name on Epoptes clients configuration with he hostname of the Fat Client running Epoptes server. 

Giacomo.

**Comment by alkisg** [comment]
I mean something like this: https://unix.stackexchange.com/questions/413948/proxying-traffic-on-local-unix-to-a-remote-unix-socket-over-ssh

See my remarks at http://irclogs.ltsp.org/?d=2019-12-30, from 10:27 and on.

**Comment by alkisg** [comment]
Btw, it's of course easy to set up a netbooted LTSP client to **run** epoptesd, to be the epoptes server.

Then of course epoptes can be run locally there.
